kyo-prelude/shared/src/main/scala/kyo2/Abort.scala

```diff
@@ -5,51 +5,50 @@ import kernel.Const
 import kernel.Effect
 import kernel.Frame
 import kyo.Tag
+import kyo2.Result.*
 import kyo2.kernel.Reducible
 import scala.annotation.targetName
 import scala.reflect.ClassTag
-import scala.util.Failure
-import scala.util.Success
-import scala.util.Try
 
-sealed trait Abort[-E] extends Effect[Const[E], Const[Unit]]
+sealed trait Abort[-E] extends Effect[Const[Failure[E]], Const[Unit]]
 
 object Abort:
 
     given eliminateAbort: Reducible.Eliminable[Abort[Nothing]] with {}
     private inline def erasedTag[E]: Tag[Abort[E]] = Tag[Abort[Any]].asInstanceOf[Tag[Abort[E]]]
 
-    inline def fail[E](inline value: E): Nothing < Abort[E] =
-        Effect.suspendMap[Any](erasedTag[E], value)(_ => ???)
+    inline def error[E](inline value: E): Nothing < Abort[E] =
+        Effect.suspendMap[Any](erasedTag[E], Error(value))(_ => ???)
 
-    inline def when[E](b: Boolean)(inline value: E)(
-        using inline tag: Tag[Abort[E]]
-    ): Unit < Abort[E] =
-        if b then fail(value)
+    inline def panic[E](ex: Throwable): Nothing < Abort[E] =
+        Effect.suspendMap[Any](erasedTag[E], Panic(ex))(_ => ???)
+
+    inline def when[E](b: Boolean)(inline value: => E): Unit < Abort[E] =
+        if b then error(value)
         else ()
 
     final class GetOps[E >: Nothing](dummy: Unit) extends AnyVal:
         inline def apply[A](either: Either[E, A]): A < Abort[E] =
             either match
                 case Right(value) => value
-                case Left(value)  => fail(value)
+                case Left(value)  => error(value)
 
-        inline def apply[A](opt: Option[A]): A < Abort[None.type] =
+        inline def apply[A](opt: Option[A]): A < Abort[Maybe.Empty] =
```

Comment:
Even if the value is an `Option`, this method new returns an `Abort[Maybe.Empty]` so we can avoid having both `Abort[None.type]` and `Abort[Maybe.Empty]` in the pending set.